torch/csrc/jit/runtime/interpreter.cpp

```diff
@@ -477,6 +479,18 @@ struct CodeImpl {
   // map from unique of nodes to register in register table
   std::unordered_map<Value*, int> value_to_reg_;
 
+  // map from operator name to trailing unnecessary inputs
+  // Example: for a schema of aten::foo.str
+  // aten::foo.str(arg0: str="default", arg1: int=0,
+  //               arg2: bool=False, arg3: float=0.0)
+  // If the usages in a graph is:
+  //    aten::foo("somestr", arg1=0, arg2=False, arg3=0.0)
+  //    aten::foo("somestr", arg1=1, arg2=False, arg3=0.0)
+  // op_to_num_unnecessary_args_["aten::foo.str"] = 2
```

Comment:
If we have another usage,
```
aten::foo("somestr", arg1=1, arg2=True, arg3=0.0)
op_to_num_unnecessary_args_["aten::foo.str"] = 1
```
Is that correct? If so, I'd suggest to use 
```
op_to_num_specified_args_["aten::foo.str"] = 3
```
The reason is, if later we update aten::foo.str and add another default arg, 
```
  // aten::foo.str(arg0: str="default", arg1: int=0,
  //               arg2: bool=False, arg3: float=0.0, arg4: bool=False)
```
In such case, the above 
```
op_to_num_unnecessary_args_["aten::foo.str"] = 1
```
in the old model cannot be used for backward compatibility. In the old runtime, it means the last argument's value, "0.0", is not specified. However, in the new runtime, it means the last argument's value, "Flase" is not specified. There is ambiguity. 
If we use 
```
op_to_num_specified_args_["aten::foo.str"] = 3
```
In old runtime we push "0.0" into the stack. In new runtime we push both "0.0" and "False" into the stack. Backward compatibility can be handled.

Reply:
makes sense. What should the map return in following case:

**schema:** 
prim::Foo(...) -> ...
**usages:** 
1. prim::Foo(arg1, arg2)
2. prim::Foo(arg1, arg2, arg3) 

Previously, we know the number of unnecessary args should be 0. But in this case, it is unclear. Maybe mobile team has to check if the schema is vararg as a special case handling? @gmagogsfm @iseeyuan

Reply:
@tugsbayasgalan As discussed offline, we don't allow default arguments in vararg ops.From mobile side we could skip vararg ops on default args.